paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkGenericCatalogFactory.java

```diff
@@ -56,14 +58,24 @@ public Set<org.apache.flink.configuration.ConfigOption<?>> optionalOptions() {
     @Override
     public FlinkGenericCatalog createCatalog(Context context) {
         CatalogFactory hiveFactory = createHiveCatalogFactory(context.getClassLoader());
-        Catalog catalog = hiveFactory.createCatalog(context);
+        Context flinkContext = createFlinkContext(context);
+        Catalog flinkCatalog = hiveFactory.createCatalog(flinkContext);
         return createCatalog(
-                context.getClassLoader(), context.getOptions(), context.getName(), catalog);
+                context.getClassLoader(), context.getOptions(), context.getName(), flinkCatalog);
+    }
+
+    @VisibleForTesting
+    public Context createFlinkContext(Context context) {
+        return new FactoryUtil.DefaultCatalogContext(
+                context.getName(),
+                OptionsUtils.convertToPropertiesPrefixKey(context.getOptions(), HIVE_PREFIX),
+                context.getConfiguration(),
+                context.getClassLoader());
```

Comment:
I think there is a problem with this modification. The key of paimon hive catalog options does not start with `hive.`

Reply:
> I think there is a problem with this modification. The key of paimon hive catalog options does not start with `hive.`

Hi @zhuangchong , thx for review. I think we're creating flink hive catalog here, not paimon hive catalog.